service/src/main/java/uk/gov/hmcts/reform/fpl/service/email/RepresentativesInbox.java

```diff
@@ -47,11 +59,66 @@
                 .collect(Collectors.toCollection(LinkedHashSet::new))
         );
 
-
         return emails;
     }
 
-    private boolean shouldSend(RepresentativeServingPreferences preference, Element<? extends WithSolicitor> element) {
+    @SuppressWarnings("unchecked")
+    public Set<?> getNonSelectedRespondentsRecipients(
+        RepresentativeServingPreferences servingPreferences,
+        CaseData caseData,
+        List<Element<Respondent>> respondentsSelected,
+        Function<Element<Representative>, ?> mapperFunction) {
+
+        Set<UUID> allRepresentativeIds = unwrapElements(caseData.getAllRespondents())
+            .stream()
+            .flatMap(respondent -> respondent.getRepresentedBy()
+                .stream().map(Element::getValue))
+            .collect(Collectors.toSet());
+
+        Set<UUID> selectedRepresentativeIds = respondentsSelected.stream()
+            .flatMap(otherElement -> otherElement.getValue().getRepresentedBy().stream().map(Element::getValue))
+            .collect(Collectors.toSet());
+
+        LinkedHashSet<Recipient> notSelectedRepresentedRespondents = (LinkedHashSet<Recipient>)
+            caseData.getRepresentativesElementsByServedPreference(servingPreferences)
+                .stream()
+                .filter(representativeElement -> allRepresentativeIds.contains(representativeElement.getId()))
+                .filter(not(representativeElement ->
+                    selectedRepresentativeIds.contains(representativeElement.getId())))
+                .map(mapperFunction)
+                .collect(Collectors.toCollection(LinkedHashSet::new));
+
+        final Set<Recipient> nonSelectedRespondents = new LinkedHashSet<>(notSelectedRepresentedRespondents);
+        if (servingPreferences == POST) {
+            nonSelectedRespondents.addAll(getNotSelectedUnrepresentedRespondents(caseData, respondentsSelected));
+        }
+        return nonSelectedRespondents;
+    }
+
+    private Set<Recipient> getNotSelectedUnrepresentedRespondents(CaseData caseData,
+                                                                  List<Element<Respondent>> selectedRespondents) {
+        List<Element<Respondent>> unrepresentedRespondents = caseData.getAllRespondents().stream()
+            .filter(respondent -> isEmpty(respondent.getValue().getRepresentedBy())
```

Comment:
just a suggestion: you can map before filter "map(Element:getValue)"

Reply:
I tried it but it returns List<Respondent> instead of List<Element<Respondent>>. I want the respondent elements to 
 compare the respondents ids in the next line

Reply:
it's cool, then.